src/pudl/transform/classes.py

```diff
@@ -252,9 +252,54 @@ def normalize_strings(col: pd.Series, params: StringNormalization) -> pd.Series:
 normalize_strings_multicol = multicol_transform_factory(normalize_strings)
 
 
+################################################################################
+# Enforce Snake Case
+################################################################################
+
+
+class EnforceSnakeCase(TransformParams):
+    """Boolean parameter for :func:`enforce_snake_case`."""
+
+    enforce_snake_case: bool
+
+
+def enforce_snake_case(
+    col: pd.Series, params: EnforceSnakeCase = EnforceSnakeCase(enforce_snake_case=True)
+) -> pd.Series:
+    """Enforce snake_case for a column.
+
+    Removes leading whitespaces, lower-cases, replaces spaces with underscore and
+    removes remaining non alpha numeric snake case values.
+
+    Args:
+        col: a column of strings.
+        params: an :class:`EnforceSnakeCase` parameter object. Default is an instance
+            of :class:`EnforceSnakeCase` where ``enforce_snake_case`` is ``True``,
+            which will enforce snake case on the ``col``. If ``False``, the column will
+            be returned unaltered.
+    """
+    if params.enforce_snake_case:
+        col = (
+            col.str.strip()
+            .str.lower()
+            .replace(
+                to_replace={
+                    r"\s{1,}": "_",
+                    r"[^a-z\d_]": "",
+                },
+                regex=True,
+            )
+        )
+    return col
+
+
+enforce_snake_case_multicol = multicol_transform_factory(enforce_snake_case)
+
 ################################################################################
 # Categorize Strings
 ################################################################################
+
+
```

Comment:
You can put comments right above the function definitions, so long as there are two blank lines separating the top of the comments from the code above -- it's not necessary to have big blocks of space around the comments on both sides.